jacodb-analysis/src/main/kotlin/org/jacodb/analysis/codegen/Generator.kt

```diff
@@ -218,21 +218,25 @@ private fun runCmd(
 }
 
 private fun checkJava(isWindows: Boolean) {
-    val javaHome = System.getenv("JAVA_HOME")
-    if (!javaHome.isNullOrEmpty()) return
-    if (System.getenv("PATH").isNullOrEmpty()) {
-        logger.error{"No PATH variable found"}
-        throw IOException("No PATH variable found")
-    }
     val canonicalJavaName = "java" + if (isWindows) ".exe" else ""
-    val javas = System.getenv("PATH").split(File.pathSeparator).filter { s: String ->
-        File(s + File.separator + canonicalJavaName).exists()
-    }
-    if (javas.isEmpty()) {
-        logger.error { "no java found in environment" }
-        throw IOException("no java found in environment")
+    val awaitJava: String
+    var javaHome = System.getenv("JAVA_HOME")
+    if (javaHome.isNullOrEmpty()) {
+        if (System.getenv("PATH").isNullOrEmpty()) {
```

Comment:
I believe you can check it buy just running `java` process and checking it exit code